src/Mod/CAM/Path/Op/Custom.py

```diff
@@ -163,28 +163,52 @@ def findGcodeFile(self, filename):
 
     def opExecute(self, obj):
         self.commandlist.append(Path.Command("(Begin Custom)"))
+        errorNumLines = []
+        errorLines = []
+        counter = 0
 
         if obj.Source == "Text" and obj.Gcode:
             for l in obj.Gcode:
-                newcommand = Path.Command(str(l))
-                self.commandlist.append(newcommand)
+                counter += 1
+                try:
+                    newcommand = Path.Command(str(l))
+                    self.commandlist.append(newcommand)
+                except ValueError:
+                    errorNumLines.append(counter)
+                    if len(errorLines) < 7:
+                        errorLines.append(f"{counter}: {str(l).strip()}")
+            if errorLines:
+                errorTitle = f"Total {len(errorNumLines)} invalid line(s) in Custom Text G-code"
+
         elif obj.Source == "File" and len(obj.GcodeFile) > 0:
             gcode_file = self.findGcodeFile(obj.GcodeFile)
 
             # could not determine the path
             if not gcode_file:
                 Path.Log.error(
-                    translate("PathCustom", "Custom file %s could not be found.") % obj.GcodeFile
+                    translate("PathCustom", f"Custom file {obj.GcodeFile} could not be found.")
                 )
-
-            with open(gcode_file) as fd:
-                for l in fd.readlines():
-                    try:
-                        newcommand = Path.Command(str(l))
-                        self.commandlist.append(newcommand)
-                    except ValueError:
-                        Path.Log.warning(translate("PathCustom", "Invalid G-code line: %s") % l)
-                        continue
+            else:
+                with open(gcode_file) as fd:
+                    for l in fd.readlines():
+                        counter += 1
+                        try:
+                            newcommand = Path.Command(str(l))
+                            self.commandlist.append(newcommand)
+                        except ValueError:
+                            errorNumLines.append(counter)
+                            if len(errorLines) < 7:
+                                errorLines.append(f"{counter}: {str(l).strip()}")
+                if errorNumLines:
+                    errorTitle = f'Total {len(errorNumLines)} invalid line(s) in Custom File G-code\n"{gcode_file}"'
+
+        if errorNumLines:
+            errorText1 = (
+                f"{errorTitle}\n\nPlease check lines: {", ".join(map(str,errorNumLines))}\n\n"
+            )
+            errorText2 = "\n".join(errorLines)
+            errorText3 = "\n..." if len(errorNumLines) > 7 else ""
+            Path.Log.warning(translate("PathCustom", f"{errorText1}{errorText2}{errorText3}"))
```

Comment:
This doesn't work. The strings in translate needs to be static, so translate the literals instead and modify them with format